Source/Parser/Expressions/ExpressionBase.cs

```diff
@@ -122,6 +123,35 @@ internal static void SkipWhitespace(PositionalTokenizer tokenizer)
 
                 tokenizer.SkipWhitespace();
             }
+
+            while (tokenizer.Match("/*"))
+            {
+                var expressionTokenizer = tokenizer as ExpressionTokenizer;
+
+                if (expressionTokenizer != null)
+                {
+                    int startLine = tokenizer.Line;
+                    int startColumn = tokenizer.Column - 2;
+                    
+                    var comment = tokenizer.ReadTo("*/");
+                    tokenizer.Advance(2);
+
+                    int endLine = tokenizer.Line;
+                    int endColumn = tokenizer.Column;
+
+                    expressionTokenizer.AddComment(new CommentExpression("/*" + comment.ToString() + "*/")
+                    {
+                        Location = new TextRange(startLine, startColumn, endLine, endColumn)
+                    });
+                }
+                else
+                {
+                    tokenizer.ReadTo("*/");
+                    tokenizer.Advance(2);
+                }
+
+                tokenizer.SkipWhitespace();
+            }
```

Comment:
This doesn't handle switching back and forth between single line comments and multiline comments:
```
// single
/* multi */
// single
/* multi */
code()
```

It matches 0 or more single, then 0 or more multi, then doesn't look for any more singles.

Reply:
Please add unit tests for this behavior. Here's some quick ones I wrote up the should be added near the existing ones in ExpressionBaseTests:
```
        [Test]
        public void TestSkipWhitespaceMultilineComment()
        {
            var tokenizer = new PositionalTokenizer(Tokenizer.CreateTokenizer("  /* multi-line\r\n     comment\r\n    */\r\nhi"));
            ExpressionBase.SkipWhitespace(tokenizer);
            Assert.That(tokenizer.NextChar, Is.EqualTo('h'));
        }

        [Test]
        public void TestSkipWhitespaceMultipleMultilineComment()
        {
            var tokenizer = new PositionalTokenizer(Tokenizer.CreateTokenizer("  /* multi-line\r\n     comment\r\n    */\r\n/* second */\r\nhi"));
            ExpressionBase.SkipWhitespace(tokenizer);
            Assert.That(tokenizer.NextChar, Is.EqualTo('h'));
        }

        [Test]
        public void TestSkipWhitespaceMultilineCommentUnclosed()
        {
            var tokenizer = new PositionalTokenizer(Tokenizer.CreateTokenizer("  /* multi-line\r\n     comment\r\nhi"));
            ExpressionBase.SkipWhitespace(tokenizer);
            Assert.That(tokenizer.NextChar, Is.EqualTo('\0'));
        }

        [Test]
        public void TestSkipWhitespaceMultilineMixedComment()
        {
            var tokenizer = new PositionalTokenizer(Tokenizer.CreateTokenizer("  // single-line\r\n/* multi-line\r\n     comment\r\n    */\r\n// second single line\r\n/* second multiline */\r\nhi"));
            ExpressionBase.SkipWhitespace(tokenizer);
            Assert.That(tokenizer.NextChar, Is.EqualTo('h'));
        }
```

Reply:
Switching back and forth between single and multiline comments should now be fixed. I also added these unit tests into ExpressionBaseTests.